woocommerce/pinterest-for-woocommerce
# Notice link click in landing page

### Changes proposed in this Pull Request:

Closes part of #232 

get_started_notice_link_click | Clicking on a text link within the notice on the Get Started page https://d.pr/i/NWRC2W  | "link_id: context: href:"
-- | -- | -- 

### Screenshots:

Pre Launch Component
![ezgif-3-bb1df3f1a0e2](https://user-images.githubusercontent.com/5908855/145851520-bc9eb550-c71f-477f-9508-a961879fd48b.gif)

Unsuported country
![ezgif-3-1ec14015acdf](https://user-images.githubusercontent.com/5908855/145851702-198d8a99-7483-47a9-84af-b011c42f1926.gif)

![Screenshot 2021-12-13 at 17 51 28](https://user-images.githubusercontent.com/5908855/145856247-f2e19055-feb7-4a40-ba15-458892af69c6.png)


### Detailed test instructions:

**Test Preparation:**

- Go to  `includes/admin/class-pinterest-for-woocommerce-admin.php` and remove your country from the list of supported countries to test the `UnsupportedCountryNotice` (it renders only if no country supported is selected). Other options is change `isAdsSupportedCountry` to true in line 265 of `assets/source/setup-guide/app/views/LandingPageApp.js`
- Go to `pinterest-for-woocommerce.php#L47` and add `beta` after the version, for example `1.0.1-beta` this is because PreLaunchNotice only renders on beta versions. Other option is to change `pluginVersion.includes( 'beta' )` to true in line 257 of `assets/source/setup-guide/app/views/LandingPageApp.js`
- Open the console and run `localStorage.setItem('debug', 'wc-admin*')` (be sure you have verbose mode activated) this is for activating the debug

**Test functionality:** 

1. Go to `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Flanding`
2. You should see the notice regarding PreLaunch and Country Unsupported rendered
3. Click on all the links in the notices
4. You should see the events `wcadmin_pfw_get_started_notice_link_click` logged with the href, link_id and context props

**Unit tests:**
This PR has unit tests and you can run them by `npm run test:js -- tracking` 

### Technical details

I used `documentationLinkProps` function by adding just the eventName param. Then, in `<PrelaunchNotice>` and `<UnsupportedCountryNotice>` I trigger the events by using that function. That way , we can avoid code duplication since both events uses the same params and logic.

Notice since I reused  `documentationLinkProps` now it not only triggers documentationLink event and also can trigger this event. So I created this issue #298 in order to adapt the naming and location of this function.  

The context string should be refactoring after  #295  is approved since it includes a nice helper function to get automatically the context from the path. 

### Changelog entry

> Tweak - Add tracking on Notice Links inside Landing Page
